cmake/DetectOptions.cmake

```diff
@@ -92,15 +92,14 @@ if(Blosc2_FOUND)
 
   set(adios2_blosc2_tgt Blosc2::Blosc2)
   if (Blosc2_VERSION VERSION_GREATER_EQUAL 2.10.1)
-    if (Blosc2_shlib_available)
-      set(adios2_blosc2_tgt Blosc2::blosc2_$<IF:$<BOOL:${ADIOS2_Blosc2_PREFER_SHARED}>,shared,static>)
+    if (Blosc2_shlib_available AND ADIOS2_Blosc2_PREFER_SHARED)
+      set(adios2_blosc2_tgt Blosc2::blosc2_shared)
     else()
```

Comment:
We will never use the `Blosc2::blosc2_static` target?

Reply:
only if it blosc2_shared does not exists or the user specified -DADIOS2_Blosc2_PREFER_SHARED=OFF